storage/src/test/java/tech/pegasys/teku/storage/client/RecentChainDataTest.java

```diff
@@ -564,6 +578,47 @@ public void commit_pruneParallelExistingBlocks() throws Exception {
     testCommitPruningOfParallelBlocks(false);
   }
 
+  @Test
+  public void getEveryRootOnChainTillSlot() {
```

Comment:
Probably should have a test case for when the block root is unknown.

Reply:
Done.